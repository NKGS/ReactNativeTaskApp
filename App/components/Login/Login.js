import React from 'react';
import { TouchableOpacity, View, Text, TextInput, TouchableWithoutFeedback, TouchableHighlight } from 'react-native';
import commonStyles, { stylesList } from '../../utils/commonStyles';

const Login = (({ register, onLoginClick }) => {
    return (
        <View style={stylesList().subView}>
            <View style={stylesList().loginTextView}>
                <Text style={commonStyles.headerTextStyle}>LOGIN</Text>
            </View>
            <View style={stylesList().loginFormView}>
                <Text style={commonStyles.mediumText}>Email</Text>
                <TextInput style={commonStyles.textInputStyle} />

                <View>
                    <Text style={commonStyles.mediumText}>Password</Text>
                </View>
                <TextInput style={commonStyles.textInputStyle} />

                <TouchableOpacity onPress={() => onLoginClick()} style={commonStyles.loginBtnView}>
                    <Text style={commonStyles.btnTextStyle}>Login</Text>
                </TouchableOpacity>
                <View style={stylesList().newUserView}>
                        <Text  onPress={() => register()} style={commonStyles.userRegisterTextStyle}>New User? Register Here</Text>
                </View>
            </View>
        </View>
    );
})


export default Login;
