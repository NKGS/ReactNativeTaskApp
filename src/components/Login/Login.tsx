import React from 'react';
import { TouchableOpacity, View, Text, TextInput, TouchableWithoutFeedback, TouchableHighlight } from 'react-native';
import commonStyles, { stylesList } from '../../utils/commonStyles';
import { RegularText } from '../Basic/Basic';

export interface Props {
    register: any,
    onLoginClick: any,
    user: any
}

const Login: React.FC<Props> = ({ user, register, onLoginClick }) => {
    return (
        <View style={stylesList().subView}>
            <View style={stylesList().loginTextView}>
                <Text style={commonStyles.headerTextStyle}>LOGIN {user ? user.name : ''}</Text>
            </View>
            <View style={stylesList().loginFormView}>
                <Text style={commonStyles.mediumText}>Email</Text>
                <TextInput style={commonStyles.textInputStyle} />

                <View>
                    <Text style={commonStyles.mediumText}>Password</Text>
                </View>
                <TextInput style={commonStyles.textInputStyle} />

                <TouchableOpacity style={stylesList().loginBtnView} testID="login-btn" onPress={() => onLoginClick()} >
                    {/* <Text style={commonStyles.btnTextStyle}>Login</Text> */}
                    <RegularText>Login</RegularText>
                </TouchableOpacity>

                <View style={stylesList().newUserView}>
                    {/* <Text onPress={() => register()} style={commonStyles.userRegisterTextStyle}>New User? Register Here</Text> */}
                    <RegularText onPress={()=> register() }>New User? Register Here</RegularText>
                </View>
            </View>
        </View>
    )
}



export default Login;
