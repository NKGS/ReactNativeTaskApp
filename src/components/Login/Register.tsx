import React from 'react';
import { StyleSheet, TouchableOpacity, View, Text, TextInput, ViewStyle, TextStyle, StyleProp } from 'react-native';
import commonStyles, { stylesList } from '../../utils/commonStyles';

export interface Props {
    login: any,
    onRegisterClick: any,
    buttonStyle?: StyleProp<ViewStyle>;
    labelStyle?: StyleProp<TextStyle>;
}

interface Styles {
    button: ViewStyle
}

const styles = StyleSheet.create<Styles>({
    button: {
      flex:1,
      alignItems:'center',
      justifyContent:'center'
    }
  });

const Register: React.FC<Props> = ({ login, onRegisterClick }):  JSX.Element => {
    return (
        <View style={stylesList().subView}>
            <View style={stylesList().loginTextView}>
                <Text style={commonStyles.headerTextStyle}>REGISTER</Text>
            </View>
            <View style={stylesList().loginFormView}>
                <Text style={commonStyles.mediumText}>Email</Text>
                <TextInput style={commonStyles.textInputStyle} />

                <View>
                    <Text style={commonStyles.mediumText}>Password</Text>
                </View>
                <TextInput style={commonStyles.textInputStyle} />

                <View>
                    <Text style={commonStyles.mediumText}>Confirm Password</Text>
                </View>
                <TextInput style={commonStyles.textInputStyle} />

                <TouchableOpacity onPress={() => onRegisterClick()} style={commonStyles.loginBtnView}>
                    <View style={styles.button}>
                        <Text>Register</Text>
                    </View>
                </TouchableOpacity>

                <View style={stylesList().newUserView}>
                    <Text style={commonStyles.mediumText}>By registering, you automatically</Text>
                    <View style={{ flexDirection: 'row' }}>
                        <Text style={commonStyles.mediumText}>accept the</Text>
                        <Text style={commonStyles.userRegisterTextStyle}>Terms & Policies</Text>
                        <Text style={commonStyles.mediumText}>of</Text>
                    </View>
                    <Text style={commonStyles.mediumText}>candy app.</Text>
                </View>
                <View style={[stylesList().newUserView, { flex: 1, justifyContent: 'flex-end', marginVertical: 30 }]}>
                    <Text style={commonStyles.userRegisterTextStyle} onPress={() => login()}>Have account? Log In</Text>
                </View>
            </View>
        </View>
    )
}

export default Register;

