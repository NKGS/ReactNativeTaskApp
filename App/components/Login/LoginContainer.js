import React, { useState } from 'react';
import {  KeyboardAvoidingView, TouchableWithoutFeedback, View, Keyboard, ScrollView } from 'react-native';
import { stylesList } from '../../utils/commonStyles';
import Login from './Login';
import Register from './Register';
import { Actions } from 'react-native-router-flux';

export default function LoginContainer() {

    const [isLoginScreen, setIsLogin] = useState(true)
    const [emailId, setEmailId] = useState('');
    const [password, setPassword] = useState('');


    onLoginClick = () => {
        Actions.push("NotesListContainer")
    }

    onRegisterClick = () => {
        alert("User addedd successfullly!!")
    }

    return (
        <KeyboardAvoidingView
            behavior={Platform.OS == "ios" ? "padding" : "height"}
            style={stylesList().container}
            enabled
            keyboardVerticalOffset={30}
        >
            <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
                <ScrollView contentContainerStyle={stylesList().scrollViewContainerStyle}>
                    <View style={stylesList().mainView}>
                        {
                         isLoginScreen ? 
                         <Login register={() => setIsLogin(false)} onLoginClick={() => this.onLoginClick()} /> : 
                         <Register login={() => setIsLogin(true)} onRegisterClick={() => this.onRegisterClick() } /> 
                        }
                    </View>
                </ScrollView>
            </TouchableWithoutFeedback>
        </KeyboardAvoidingView>
    );
}

