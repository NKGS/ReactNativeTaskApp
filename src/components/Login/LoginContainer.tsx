import React, { useState, useEffect } from 'react';
import { Text, Platform, PermissionsAndroid, KeyboardAvoidingView, TouchableWithoutFeedback, View, Keyboard, ScrollView } from 'react-native';
import { stylesList } from '../../utils/commonStyles';
import Login from './Login';
import Register from './Register';
import { useDispatch, useSelector } from 'react-redux';
import { loginAction } from '../../redux/actions/login-actions';
import { startApp } from '../../App';
// import { useNavigation } from '@react-navigation/native';
//emailId={emailId} password={password} setEmailId={(val) => setEmailId(val) } setPassword={(val) => setPassword(val)} 

const LoginContainer  = (props) => {

    const dispatch = useDispatch();

    const user = useSelector(state => state.login.user);

    const [isLoginScreen, setIsLogin] = useState(true)  
    //const {navigate} = useNavigation();
    const [emailId, setEmailId] = useState('');
    const [password, setPassword] = useState('');

    const onLoginClick = () => {
      dispatch(loginAction('mk'))
      console.log('asas')
    }

    const onRegisterClick = () => {
        alert("User addedd successfullly!!")
    }
    
    useEffect(() => {
        if (user && user.name) {
            startApp();
        }
    })

    return (
        <KeyboardAvoidingView
            behavior={Platform.OS == "ios" ? "padding" : "height"}
            style={stylesList().container}
            enabled
            keyboardVerticalOffset={30}
        >
            <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
                <ScrollView contentContainerStyle={stylesList().scrollViewContainerStyle}>
                    <View style={stylesList().mainView}>
                        {
                         isLoginScreen ? 
                         <Login user={user} register={() => setIsLogin(false)} onLoginClick={() => onLoginClick()} /> : 
                         <Register login={() => setIsLogin(true)} onRegisterClick={() => onRegisterClick() } /> 
                        }
                    </View>
                </ScrollView>
            </TouchableWithoutFeedback>
        </KeyboardAvoidingView>
    );
}

// function mapStateToProps(state) {
//     return {
//         books: state.Books
//     }
// }
// function mapDispatchToProps(dispatch) {
//     return {
//         dispacthLogin: (payload)=> dispatch({type: 'LOGIN', payload})
//     }
// }

// export const LoginContainer = connect(mapStateToProps, mapDispatchToProps)(LoginContainers)
export default LoginContainer;