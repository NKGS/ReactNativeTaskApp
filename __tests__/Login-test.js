import React from 'react';
import renderer from 'react-test-renderer';
import Login from './../src/components/Login/Login';
import Register from '../src/components/Login/Register';

it('renders Registrer correctly', () => {
    beforeAll(() => { 
      jest.mock('@react-native-community/async-storage');
    })
    let register = renderer.create(<Register />).toJSON();
    expect(register).toMatchSnapshot();
  });

  it('renders Login correctly', () => {
    beforeAll(() => { 
      jest.mock('@react-native-community/async-storage');
    })
    let login = renderer.create(<Login />).toJSON();
    console.log(login)
    expect(login).toMatchSnapshot();
  });
  