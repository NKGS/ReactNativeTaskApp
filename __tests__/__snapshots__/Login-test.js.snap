// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.2,
        "justifyContent": "flex-end",
        "marginBottom": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3B3B3B",
          "fontSize": 20,
        }
      }
    >
      LOGIN 
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 0.8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3B3B3B",
          "fontSize": 12,
        }
      }
    >
      Email
    </Text>
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginVertical": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View>
      <Text
        style={
          Object {
            "color": "#3B3B3B",
            "fontSize": 12,
          }
        }
      >
        Password
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginVertical": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 2,
          "flex": 1,
          "height": 58,
          "justifyContent": "center",
          "marginVertical": 18,
          "maxHeight": 50,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 10,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "borderBottomWidth": 1,
            "color": "#3B3B3B",
            "fontSize": 13,
          }
        }
      >
        New User? Register Here
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders Registrer correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.2,
        "justifyContent": "flex-end",
        "marginBottom": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3B3B3B",
          "fontSize": 20,
        }
      }
    >
      REGISTER
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 0.8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3B3B3B",
          "fontSize": 12,
        }
      }
    >
      Email
    </Text>
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginVertical": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View>
      <Text
        style={
          Object {
            "color": "#3B3B3B",
            "fontSize": 12,
          }
        }
      >
        Password
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginVertical": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View>
      <Text
        style={
          Object {
            "color": "#3B3B3B",
            "fontSize": 12,
          }
        }
      >
        Confirm Password
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 1,
          "marginVertical": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8CA31",
          "borderRadius": 8,
          "borderWidth": 2,
          "flex": 1,
          "height": 58,
          "justifyContent": "center",
          "marginVertical": 18,
          "maxHeight": 50,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text>
          Register
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3B3B3B",
            "fontSize": 12,
          }
        }
      >
        By registering, you automatically
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3B3B3B",
              "fontSize": 12,
            }
          }
        >
          accept the
        </Text>
        <Text
          style={
            Object {
              "borderBottomWidth": 1,
              "color": "#3B3B3B",
              "fontSize": 13,
            }
          }
        >
          Terms & Policies
        </Text>
        <Text
          style={
            Object {
              "color": "#3B3B3B",
              "fontSize": 12,
            }
          }
        >
          of
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#3B3B3B",
            "fontSize": 12,
          }
        }
      >
        candy app.
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 10,
          },
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginVertical": 30,
          },
        ]
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "borderBottomWidth": 1,
            "color": "#3B3B3B",
            "fontSize": 13,
          }
        }
      >
        Have account? Log In
      </Text>
    </View>
  </View>
</View>
`;
